xdsl/dialects/arm/register.py

```diff
@@ -36,6 +38,27 @@ def infinite_register_prefix(cls):
         return "inf_"
 
 
+@irdl_attr_definition
+class NEON128RegisterType(ARMRegisterType):
+    """
+    A 128-bit NEON ARM register type.
+    """
+
+    name = "arm_neon.neon128reg"
```

Comment:
Shouldn't this be in the arm_neon dialect folder/file?

Reply:
I have moved this into arm_neon.py